hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -116,18 +114,16 @@ public void onListenerRegister() {
 
     private void waitForPartitionsFetchedOnce() {
         while ((partitions.isEmpty() || partitionCount == 0) && client.getConnectionManager().isAlive()) {
-            if (isClusterFormedByOnlyLiteMembers()) {
-                throw new NoDataMemberInClusterException(
-                        "Partitions can't be assigned since all nodes in the cluster are lite members");
-            }
             ClientMessage requestMessage = ClientGetPartitionsCodec.encodeRequest();
             ClientInvocationFuture future = new ClientInvocation(client, requestMessage, null).invokeUrgent();
             try {
                 ClientMessage responseMessage = future.get();
-                ClientGetPartitionsCodec.ResponseParameters response =
-                        ClientGetPartitionsCodec.decodeResponse(responseMessage);
-                processPartitionResponse(response.partitions,
-                        response.partitionStateVersion, response.partitionStateVersionExist);
+                ClientGetPartitionsCodec.ResponseParameters response = ClientGetPartitionsCodec.decodeResponse(responseMessage);
+                if (!processPartitionResponse(response.partitions,
+                        response.partitionStateVersion, response.partitionStateVersionExist)) {
+                    throw new NoDataMemberInClusterException(
+                            "Cluster could not initialize the partition table ");
+                }
```

Comment:
Previous code has the check for lite members and I do not see the corresponding check here? Is this intentional?